docker/Dockerfile.py

```diff
@@ -141,6 +145,15 @@ def heredoc(s):
     # <https://github.com/apptainer/singularity/issues/6113#issuecomment-901897566>).
     #
     # So we need to make sure to install a downgraded squashfs first.
+    #
+    # TODO: Singularity has since resolved this on their end
+    # https://github.com/sylabs/singularity/pull/267
+    # This works by checking that the UID of the caller is not root
+    # In a Kubernetes pod, the default setup will have UID 0 even if the pod is unprivileged
+    # https://github.com/sylabs/singularity/issues/2727
+    # It is possible to avoid this by changing the user of the pod (such as runAsUser: 1000)
+    # but this may cause permission issues, ex: changed read/write permissions
+    # so the downgraded squashfs stays, but options for updated squashfs are possible
```

Comment:
What options do you mean?

Reply:
I did try to see if Singularity would work with an updated squashfs-tools package by running under a user that isn't UID 0. Singularity did run, but the user won't have sufficient permissions for things such as writing files. I haven't followed through yet as I'm not too sure about the best way to give the user permissions, since existing normal Toil workflows could be writing to root owned directories currently. I could probably simply chmod certain directories that I think are necessary (such as `/` and other directories that aren't kernel/OS related).